### PR TITLE
make flags more intuitive

### PR DESCRIPTION
**Pull request by axel-h**
- `disabled` -> `no_hw_test`
- `simulation_binary` -> `has_simulation` as boolean

### CODE REVIEW

File: sel4test-sim/builds.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ build-filter:
 #      - arch: [x86, riscv]
 #        compiler: [gcc]
 #        mode: [32]
-#        simulation_binary:
+#        has_simulation:
```
Comment (lsf37) [inline sel4test-sim/builds.yml:68]: To make it clearer, let's bring the example in sync with the use above.

```suggestion
#        has_simulation: true
```

File: seL4-platforms/platforms.yml [hunk 2]
```diff
@@ -16,47 +16,47 @@ architectures: [x86, arm, riscv]
...
 #     platform: name; required
 #     image_platform: name; optional
-#     simulation_binary: string; optional
+#     has_simulation: boolean; optional; (null = false); platform is a simulation
 #     march: string; optional
 #     req: list of string; optional; names of test boards in machine queue
-#     disabled: optional; (null = false); no hw test on this platform
-#     no_hw_build: optional; (null = false); no hw build for this platform
+#     no_hw_test: optional; (null = false); no hw test on this platform
+#     no_hw_build: optional; (null = false); no hw build for this platform (implies no_hw_test then)
```
Comment (lsf37) [inline seL4-platforms/platforms.yml:23]:
```suggestion
#     no_hw_build: optional; (null = false); no hw build for this platform (implies no_hw_test)
```

**Review by lsf37 — APPROVED** [review]
I think this makes sense. The `simulation_binary` setting is a relic from very old times and at best confusing, and `no_hw_test` is better in line with the rest of the terminology and the labels we use on repos.

@june-andronick if you're looking into what is tested, this PR will change the names of the keys you have to check for.

File: sel4test-sim/builds.yml (hunk 1 above)
Comment (axel-h) [reply sel4test-sim/builds.yml]: done

File: seL4-platforms/platforms.yml (hunk 2 above)
Comment (axel-h) [reply seL4-platforms/platforms.yml]: done

**Pull request merged**